### ISSUE TITLE
Python 3 support - failure on installation

### ISSUE THREAD

**Issue by tracek** [melizalab/libtfr#7]
I am trying to install the lib on Python 3.5.2 on Linux (xubuntu 16.04). Here's the output:

```
pip install libtfr
Collecting libtfr
  Using cached libtfr-2.0.0.tar.gz
    Complete output from command python setup.py egg_info:
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/tmp/pip-build-5mx024u2/libtfr/setup.py", line 54, in <module>
        compiler_settings['include_dirs'].add(numpy.get_include())
    AttributeError: 'list' object has no attribute 'add'
```

`compiler_settings['include_dirs'].add(numpy.get_include())` makes me think that the code has never been tested on Python 3.5. Fixing is easy, what worries me is how far I would have to go.

After
```
compiler_settings['include_dirs'] = numpy.get_include()
compiler_settings['libraries'] = 'lapack'
```

Here's what I get:

```
pip install .
Processing /home/tracek/Projects/libtfr
Installing collected packages: libtfr
  Running setup.py install for libtfr ... error
    Complete output from command /home/tracek/venv/birdman-py3/bin/python3 -u -c "import setuptools, tokenize;__file__='/tmp/pip-6pr_gw93-build/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record /tmp/pip-4t349dmx-record/install-record.txt --single-version-externally-managed --compile --install-headers /home/tracek/venv/birdman-py3/include/site/python3.5/libtfr:
    running install
    running build
    running build_ext
    skipping 'libtfr.c' Cython extension (up-to-date)
    building 'libtfr' extension
    creating build
    creating build/temp.linux-x86_64-3.5
    x86_64-linux-gnu-gcc -pthread -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -g -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -I/ -Ih -Io -Im -Ie -I/ -It -Ir -Ia -Ic -Ie -Ik -I/ -Iv -Ie -In -Iv -I/ -Ib -Ii -Ir -Id -Im -Ia -In -I- -Ip -Iy -I3 -I/ -Il -Ii -Ib -I/ -Ip -Iy -It -Ih -Io -In -I3 -I. -I5 -I/ -Is -Ii -It -Ie -I- -Ip -Ia -Ic -Ik -Ia -Ig -Ie -Is -I/ -In -Iu -Im -Ip -Iy -I/ -Ic -Io -Ir -Ie -I/ -Ii -In -Ic -Il -Iu -Id -Ie -I/usr/include/python3.5m -I/home/tracek/venv/birdman-py3/include/python3.5m -c tfr.c -o build/temp.linux-x86_64-3.5/tfr.o
    cc1: note: obsolete option -I- used, please use -iquote instead
    cc1: error: -I- specified twice
    cc1: note: obsolete option -I- used, please use -iquote instead
    error: command 'x86_64-linux-gnu-gcc' failed with exit status 1
    
    ----------------------------------------
Command "/home/tracek/venv/birdman-py3/bin/python3 -u -c "import setuptools, tokenize;__file__='/tmp/pip-6pr_gw93-build/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record /tmp/pip-4t349dmx-record/install-record.txt --single-version-externally-managed --compile --install-headers /home/tracek/venv/birdman-py3/include/site/python3.5/libtfr" failed with error code 1 in /tmp/pip-6pr_gw93-build/
```

Error: cc1: error: -I- specified twice

Any hints?

Cheers,
Lucas

**Comment by suntracks** [comment]
@dmeliza Would there any plans to fix this for python 3.6 and above ?
I dont the see the above workaround working for py3


> This is a problem with pkgconfig (the Python package that interfaces with pkg-config), which used to store its data in sets but now does so in lists. Fundamentally, the issue is that pkgconfig is required for setup.py to run, but can't be installed as a dependency by setuptools if the code that generates the extension module fails to run before setup() is called. I know there is a way to get around this, but I don't have the time to implement it now.
> 
> You should be able to get the module to install as long as you `pip install -r requirements.txt` first (if installing from source) or `pip install pkgconfig>=1.2` if installing from pypi.

**Comment by dmeliza** [comment]
This should be fixed now that we are using pep517 builds - pkgconfig will be pulled in prior to the build.
